src/clr_bundle_ls.c

```diff
@@ -195,29 +195,25 @@ int bundle_list_main(int argc, char **argv)
 	}
 
 	if (cmdline_option_deps != NULL) {
-		exit(show_included_bundles(cmdline_option_deps));
-	}
-
-	if (cmdline_option_all && cmdline_option_has_dep != NULL) {
-		exit(show_bundle_reqd_by(cmdline_option_has_dep, true));
+		ret = show_included_bundles(cmdline_option_deps);
 	} else if (cmdline_option_has_dep != NULL) {
-		exit(show_bundle_reqd_by(cmdline_option_has_dep, false));
+		ret = show_bundle_reqd_by(cmdline_option_has_dep, cmdline_option_all);
 	} else if (cmdline_option_all) {
-		exit(list_installable_bundles());
-	}
-
-	ret = list_local_bundles();
+		ret = list_installable_bundles();
+	} else {
+		ret = list_local_bundles();
+
+		current_version = get_current_version(path_prefix);
+		if (current_version < 0) {
+			fprintf(stderr, "Error: Unable to determine current OS version\n");
+			v_lockfile(lock_fd);
+			return ECURRENT_VERSION;
+		}
 
-	current_version = get_current_version(path_prefix);
-	if (current_version < 0) {
-		fprintf(stderr, "Error: Unable to determine current OS version\n");
-		v_lockfile(lock_fd);
-		return ECURRENT_VERSION;
+		/* Should this be a different command */
```

Comment:
Ha, I know you are just moving code, so no action, but yes, yes this should be a different command.

Hey, it actually is, you added the `swupd info` command that shows exactly this.

Reply:
As long as you adding miscellaneous improvements, just remove this print in another commit.